src/doc/version-switcher.js

```diff
@@ -0,0 +1,269 @@
+(function() {
+
+let CURRENT_VERSION = -1;
+
+function get_current_version() {
+    if (CURRENT_VERSION !== -1) {
+        return CURRENT_VERSION;
+    }
+    const now = Date.now();
+    // Month is 0-indexed.
+    // First release of Rust, 15 may 2015.
+    const first_release = new Date(2015, 4, 15);
+    const diff_time = Math.abs(now - first_release);
+    const nb_days = Math.ceil(diff_time / (1000 * 60 * 60 * 24));
+    const nb_weeks = nb_days / 7;
+    CURRENT_VERSION = Math.floor(nb_weeks / 6);
+    return CURRENT_VERSION;
```

Comment:
This doesn't account for patch versions. Also, it seems fragile to compute the version from the date. E.g., what if for some reason Rust had to skip or delay a release?

I think a decent solution would be to use `https://blog.rust-lang.org/releases.json` and extract the number from the title or URL.